Improve TIMER

datagrok opened the pull request:
Dear @audiodude, @selfsame, @fonorobert, @piersb, @cardboard, @william42, @kgrover, @bigjust, and @sashahart,

I present, for v̧our kind consideration, a new TIMER. Please give it v̧our :+1:s

[Here is a link to the markdown-rendered rule](https://github.com/datagrok/tiny-nomic/blob/68c2466218dd3cbe80962205fd79efcc42d003cc/mutable_rules/317_Temporal-Initiated-Motion-Every-Rule.md)

The intent behind these changes:

- Enable plav̧ers to participate in the game when their schedule allows.

- Preserve mechanisms which ensure each plav̧er gets an opportunitv̧ to have their proposed rule changes influence the game.

- Preserve mechansims which keep the game moving if one or manv̧ plav̧ers are unresponsive.

If mv̧ suspicion is correct then manv̧ of our noncommunicative plav̧ers will be able to find time to participate when thev̧ are not boxed into limited time windows for doing so.

I hope these changes will make the game feel like it is moving faster while at the same time giving plav̧ers more time for inventing rules and discussing changes.

I hope the game becomes less about waiting for timers and reacting to deadlines than about building rules good enough to pursuade others to vote for them.

The effect of this rule should be:

- Plav̧ers are still allotted 48 hours minimum to come up with a rule before forfeiting their turn. in practice, thev̧ should feel like thev̧ have _more_ time since thev̧ don't have to wait for their turn to begin to post or get votes on their proposal, and thev̧ don't have to actuallv̧ be present during their turn for an acceptable proposal to be merged.

- Plav̧ers are still allotted 24 hours for discussion and 24 hours for voting, but mav̧ accelerate these time blocks bv̧ discussing and voting earlv̧.

- The game will not run autonomouslv̧, causing plav̧ers to forefeit their turns one after another forever with no actions. If manv̧ plav̧ers are unresponsive, it still takes an active plav̧er to perform the steps to move the game along. Corollarv̧: if v̧ou _would have_ forefieted v̧our turn recentlv̧ but nobodv̧ has been active enough to notice, then in effect, v̧ou can take v̧our turn immediatelv̧ with no penaltv̧.

This rule need not establish precedence over _Parts of a turn_, bv̧ interpreting `proposing one rule-change and having it voted on` to refer to the proposal selection and vote evaluation process that it defines, which satisfies its criteria bv̧ selecting one out of possiblv̧ manv̧ rule-changes while disallowing anv̧ chance to merge without other plav̧ers having voted upon it.

This rule need not establish precedence over _Votes Per Plav̧er_, because although plav̧ers mav̧ comment with a vote anv̧ number of times, thev̧ mav̧ influence the rule-change acceptance bv̧ onlv̧ one vote per tallv̧.

Plav̧ers mav̧ push more commits to a PR to improve them after discussion. We don't want plav̧ers to "sneak changes in" that wav̧, so everv̧bodv̧ has to re-vote when a PR gets changed, and the time windows for discussion and voting restart with each commit. (This mav̧ make a rule unacceptable during a turn, and have to wait for the next turn.) But we allow a voter to explicitlv̧ state something like "this vote applies regardless of anv̧ future changes to this PR :+1:".

Just as we have been so far to allow TIMER to operate, we Orwellian-lv̧ interpret a plav̧er who, through non-participation, implicitlv̧ votes "abstain" (or "against"), as "participation" sufficient to satisfv̧ _Eligible Voters_:

> Everv̧ plav̧er is an eligible voter. Everv̧ eligible voter must participate in everv̧ vote on rule-changes.

A contentious rule proposal mav̧ exist through multiple rounds without causing point loss, if the proposing plav̧er also proposes less-contentious rule changes within the required time limits. (The first of a set of PRs that passes the vote is the one accepted for the turn.)

datagrok committed:
improve TIMER

datagrok committed:
Improve logical description of acceptability procedure

datagrok committed:
Improve GFM formatting

piersb commented:
Complicated. But better than the ghostly wasteland wherein we currently reside.
👍

selfsame commented:
fixes all of TIMERs pain points  :+1:
Also: @piersb has used the forbidden character twice, and @datagrok once, and will be penalized

piersb commented:
CURSES!

> On 28 Jul 2017, at 13:38, Joseph Parker <notifications@github.com> wrote:
> 
> fixes all of TIMERs pain points 👍
> Also: @piersb <https://github.com/piersb> has used the forbidden character twice, and @datagrok <https://github.com/datagrok> once, and will be penalized
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/audiodude/tiny-nomic/pull/67#issuecomment-318641021>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABWnVCjgTnkzJsv1drak_rCjCIJPmjQFks5sSdYugaJpZM4Ol_0e>.
> 

--
Piers Beckley
+44 (0) 79 5257 6693
piers.beckley@fatpigeons.com <mailto:piers.beckley@fatpigeons.com>
skype: piers.beckley

datagrok commented:
Argh, I missed one!

audiodude commented:
I would say that @datagrok has multiple violations of VOWEL in their pull request because of:

> where 'use' signifies inclusion of its written character, or alternate characters suggestive of its visual form

audiodude commented:
Could you please clarify point 3 of the acceptability criteria? It seems really complex, especially the part about players who would have normally abstained being counted as votes against in this one weird situation.

Also:

> Once a proposal is accepted their turn ends and the next plav̧er's begins.

Do you mean "once an acceptable proposal has been selected"?

bigjust commented:
@audiodude violated 314 6 times

datagrok commented:
@audiodude I think I'll just remove point 3.

It existed for the situation where someone has obtained a majority (thus, if nobody changes their vote, is sure to be accepted) and doesn't want to wait for the remaining votes to come in to merge and end their turn. The tricky part with abstains vs. nays was to prevent someone from voting +1 on their own proposal with no other votes and immediately declaring it accepted with 100% approval (and 100% reward points).

But with proposals that can exist across many turns before merge, I don't think that feature is worth the additional complexity anymore.

> > Once a proposal is accepted their turn ends and the next plav̧er's begins.
>
> Do you mean "once an acceptable proposal has been selected"?

No,
- "accepted" means "merged" when playing on GitHub,
- "selected" means "declared to be the proposal that gets accepted for that turn, in the situation where multiple are in an acceptable state"
- "acceptable" means they have obtained a sufficient number of +1 votes

Does that clear up that part?

I'll push a commit after work to clarify this a bit, and to remove acceptability clause 3.

> I would say that @datagrok has multiple violations of VOWEL in their pull request because of:
> > where 'use' signifies inclusion of its written character, or alternate characters suggestive of its visual form

Yikes. I misremembered the rule as only applying to a single suggestive character, so I thought I was safe with a v + combining cedilla.

Welp. Bring on the penalty points.

![oirp2](https://user-images.githubusercontent.com/569728/28733540-80834eba-7391-11e7-8545-cb1473f964d5.jpg)

I've got a future proposal to deal with that if nobody else gets there first 😈

bigjust commented:
I count 19 violations of beleaguered rule 314

> On Jul 28, 2017, at 3:42 PM, Michael F. Lamb <notifications@github.com> wrote:
> 
> @audiodude <https://github.com/audiodude> I think I'll just remove point 3.
> 
> It existed for the situation where someone has obtained a majority (thus, if nobody changes their vote, is sure to be accepted) and doesn't want to wait for the remaining votes to come in to merge and end their turn. The tricky part with abstains vs. nays was to prevent someone from voting +1 on their own proposal with no other votes and immediately declaring it accepted with 100% approval (and 100% reward points).
> 
> But with proposals that can exist across many turns before merge, I don't think that feature is worth the additional complexity anymore.
> 
> Once a proposal is accepted their turn ends and the next plav̧er's begins.
> 
> Do you mean "once an acceptable proposal has been selected"?
> 
> No,
> 
> "accepted" means "merged" when playing on GitHub,
> "selected" means "declared to be the proposal that gets accepted for that turn, in the situation where multiple are in an acceptable state"
> "acceptable" means they have obtained a sufficient number of +1 votes
> Does that clear up that part?
> 
> I'll push a commit after work to clarify this a bit, and to remove acceptability clause 3.
> 
> I would say that @datagrok <https://github.com/datagrok> has multiple violations of VOWEL in their pull request because of:
> 
> where 'use' signifies inclusion of its written character, or alternate characters suggestive of its visual form
> 
> Yikes. I misremembered the rule as only applying to a single suggestive character, so I thought I was safe with a v + combining cedilla.
> 
> Welp. Bring on the penalty points.
> 
>  <https://user-images.githubusercontent.com/569728/28733540-80834eba-7391-11e7-8545-cb1473f964d5.jpg>
> I've got a future proposal to deal with that if nobody else gets there first 😈
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/audiodude/tiny-nomic/pull/67#issuecomment-318744821>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFQAcpD9m5kzbKOxtkh71taSh2hVQ6Uks5sSjmcgaJpZM4Ol_0e>.
>

piersb commented:
Xeah, something tells me 314's not long for this world.

Sent from my iPhone

> On 28 Jul 2017, at 20:56, Justin Caratzas <notifications@github.com> wrote:
> 
> I count 19 violations of beleaguered rule 314
> 
> > On Jul 28, 2017, at 3:42 PM, Michael F. Lamb <notifications@github.com> wrote:
> > 
> > @audiodude <https://github.com/audiodude> I think I'll just remove point 3.
> > 
> > It existed for the situation where someone has obtained a majority (thus, if nobody changes their vote, is sure to be accepted) and doesn't want to wait for the remaining votes to come in to merge and end their turn. The tricky part with abstains vs. nays was to prevent someone from voting +1 on their own proposal with no other votes and immediately declaring it accepted with 100% approval (and 100% reward points).
> > 
> > But with proposals that can exist across many turns before merge, I don't think that feature is worth the additional complexity anymore.
> > 
> > Once a proposal is accepted their turn ends and the next plav̧er's begins.
> > 
> > Do you mean "once an acceptable proposal has been selected"?
> > 
> > No,
> > 
> > "accepted" means "merged" when playing on GitHub,
> > "selected" means "declared to be the proposal that gets accepted for that turn, in the situation where multiple are in an acceptable state"
> > "acceptable" means they have obtained a sufficient number of +1 votes
> > Does that clear up that part?
> > 
> > I'll push a commit after work to clarify this a bit, and to remove acceptability clause 3.
> > 
> > I would say that @datagrok <https://github.com/datagrok> has multiple violations of VOWEL in their pull request because of:
> > 
> > where 'use' signifies inclusion of its written character, or alternate characters suggestive of its visual form
> > 
> > Yikes. I misremembered the rule as only applying to a single suggestive character, so I thought I was safe with a v + combining cedilla.
> > 
> > Welp. Bring on the penalty points.
> > 
> > <https://user-images.githubusercontent.com/569728/28733540-80834eba-7391-11e7-8545-cb1473f964d5.jpg>
> > I've got a future proposal to deal with that if nobody else gets there first 😈
> > 
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub <https://github.com/audiodude/tiny-nomic/pull/67#issuecomment-318744821>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFQAcpD9m5kzbKOxtkh71taSh2hVQ6Uks5sSjmcgaJpZM4Ol_0e>.
> > 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

datagrok committed:
i'm a filthy y-user, come at me bro

i misremembered the rule so i wasn't safe hiding behind v + combining
cedilla. since i'm going to get hit with the penalty anyway, might as
well make the rule use the real unfiltered forbidden letter. the good
stuff.

datagrok committed:
remove confusing acceptability clause 3

datagrok committed:
more language clarification

datagrok commented:
@audiodude I made some improvements to the language, and removed acceptability clause 3. I tried to ensure that none of the other logic changed. [link to new text](https://github.com/datagrok/tiny-nomic/blob/287a472e3314b6e335666f48a80aa40a1619d7e4/mutable_rules/317_Temporal-Initiated-Motion-Every-Rule.md)

While working on this update I became distressed that the turn organization is still way too complex. Even so, this should make it easier to develop new rules, and I'll follow up with another change to simplify things.

I hope everyone finds it satisfactory.

@fonorobert, @Cardboard, @william42, @kgrover, and @sashahart, we have about 24 hours remaining to register your vote for this rule, which hopefully will make future rules-making less fraught with such deadlines.

audiodude commented:
I vote yes for the pure bloody thoroughness and complexity of this: 👍

sashahart commented:
:+1: more law more chaos
whi not

datagrok commented:
Oh right I need to do this too :+1:

fonorobert commented:
a bit complex, but i am in favor of making gameplai more asinchronous. i vote in favor 👍

bigjust commented:
👍 I think i understand

datagrok commented:
- [x] audiodude
- [x] selfsame
- [x] fonorobert
- [x] piersb
- [ ] cardboard (abstain)
- [ ] william42 (abstain)
- [ ] kgrover (abstain)
- [x] datagrok
- [x] bigjust
- [x] sashahart

The TIMER has expired for voting and I believe the proposal passes! Thank you everybody.

@bigjust you're up next, but with the merge of this rule now anyone may create a PR at any time.
